src/Tmds.Ssh/PrivateKeyCredential.cs

```diff
@@ -2,15 +2,21 @@
 // See file LICENSE for full license details.
 
 using System;
+using System.Text;
 
 namespace Tmds.Ssh;
 
-public sealed class PrivateKeyCredential : Credential
+public sealed partial class PrivateKeyCredential : Credential
 {
     internal string FilePath { get; }
+    internal byte[] Passphrase { get; }
 
-    public PrivateKeyCredential(string path)
+    public PrivateKeyCredential(string path) : this(path, null)
+    { }
+
+    public PrivateKeyCredential(string path, string? passphrase)
     {
         FilePath = path ?? throw new ArgumentNullException(nameof(path));
+        Passphrase = string.IsNullOrEmpty(passphrase) ? Array.Empty<byte>() : Encoding.UTF8.GetBytes(passphrase);
```

Comment:
Let's store this as a string (same type as passed to the constructor).

Reply:
I chose a `byte[]` so that the parser code didn't need to do the encoding itself. It could also be possible to use a passphrase of non-UTF8 bytes if an overload was added to accept `byte[]` alongside a string.

Reply:
I don't think we should account for non encode-able passwords.

The conversion to bytes is cheap enough to move it to the parser, and have the `Credential` class as dumb as possible. The credential classes are simple data holders that expose the constructor arguments as internal properties.